cmd/contour/serve.go

```diff
@@ -535,8 +537,18 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		return s.Serve(l)
 	})
 
-	// step 15. GO!
+	// step 15. Setup SIGTERM handler
+	g.Add(func(<-chan struct{}) error {
```

Comment:
This worker won't shut down until it receives a signal. Patch in the change from #1361 and check to see if this worker blocks shutdown of the group.